### PR TITLE
Added new unit test for cv::initInverseRectificationMap()

### PR DESCRIPTION
**Pull request by IanMaquignaz**
Added new unit test for cv::initInverseRectificationMap()

Replaces unit test from #20165 and completes the updating of cv::initInverseRectificationMap() started in #20247

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by IanMaquignaz** [comment]
Included sample distortion coefficients from a DSLR camera and a Point Grey camera. Current parameters pass using an image of `size_w_h(512 + 3, 512)`, through note the error can exceed the `1e-3` threshold if image size is increased to be larger. 

```
double d[5]={ // Non-zero distortion from a Point Grey Camera
        -3.4134571357400023e-03, 2.9733267766101856e-03, // K1, K2
        3.6653586399031184e-03, -3.1960714017365702e-03, // P1, P2
        0. // K3
    };
```

File: modules/calib3d/test/test_undistort.cpp [hunk 1]
```diff
@@ -1807,4 +1807,79 @@ TEST(Calib3d_initUndistortRectifyMap, regression_14467)
     EXPECT_LE(cvtest::norm(dst, mesh_uv, NORM_INF), 1e-3);
 }
 
+TEST(Calib3d_initInverseRectificationMap, regression_14467)
```
Comment (alalek) [inline modules/calib3d/test/test_undistort.cpp:1810]: > regression_14467

Please change this name

Comment (alalek) [inline modules/calib3d/test/test_undistort.cpp:1810]: > regression_14467

Please change this name

File: modules/calib3d/test/test_undistort.cpp [hunk 2]
```diff
@@ -1807,4 +1807,79 @@ TEST(Calib3d_initUndistortRectifyMap, regression_14467)
...
+{
+    Size size_w_h(512 + 3, 512);
```
Comment (alalek) [inline modules/calib3d/test/test_undistort.cpp:1812]: It makes sense to use a really different width/height. Use some regular resolution like 640x480 or 1280x720 (camera matrix should be updated too)

File: modules/calib3d/test/test_undistort.cpp [hunk 3]
```diff
@@ -1807,4 +1807,79 @@ TEST(Calib3d_initUndistortRectifyMap, regression_14467)
...
+    double d[5]={ // Non-zero distortion from a Point Grey Camera
+        -3.4134571357400023e-03, 2.9733267766101856e-03, // K1, K2
+        3.6653586399031184e-03, -3.1960714017365702e-03, // P1, P2
+        0. // K3
+    };
+    Mat _d(1, 5, CV_64F, d);
+
+    // Rotation
+    //double R[9]={1., 0., 0., 0., 1., 0., 0., 0., 1.}; // Identity transform (none)
+    double R[9]={1., 0.05, 0., 0.05, 1.0, 0.005, 0.005, 0., 1.}; // Random transform
```
Comment (alalek) [inline modules/calib3d/test/test_undistort.cpp:1840]: valid transforms have some idempotent properties, like det(rotation) = 1 or -1

File: modules/calib3d/test/test_undistort.cpp (hunk 1 above)
Comment (IanMaquignaz) [reply modules/calib3d/test/test_undistort.cpp:1810]: fixed. Assumed number is arbitrary and changed it to `regression_14468`.

Comment (IanMaquignaz) [reply modules/calib3d/test/test_undistort.cpp]: Assumed this is a duplicate of the comment above.

File: modules/calib3d/test/test_undistort.cpp (hunk 2 above)
Comment (IanMaquignaz) [reply modules/calib3d/test/test_undistort.cpp]: Updated intrinsics and extrinsics specification to use experimentally collected values. 
```
Size size_w_h(1280, 800);
```
```
// Camera Matrix
double k[9]={
    1.5393951443032472e+03, 0., 6.7491727003047140e+02,
    0., 1.5400748240626747e+03, 5.1226968329123963e+02,
    0., 0., 1.
};
```

```
double d[5]={ // Non-zero distortion
    -3.4134571357400023e-03, 2.9733267766101856e-03, // K1, K2
    3.6653586399031184e-03, -3.1960714017365702e-03, // P1, P2
    0. // K3
};
```

Note, K1 and K2 have been increased from `-3.4134571357400023e-01, 2.9733267766101856e-01` to meet error threshold of ` 2e-1`

File: modules/calib3d/test/test_undistort.cpp (hunk 3 above)
Comment (IanMaquignaz) [reply modules/calib3d/test/test_undistort.cpp]: See above comment

**Comment by IanMaquignaz** [comment]
It seems someone added a shell script which tries `curl -s https://api.github.com/repos/IanMaquignaz/opencv_extra/branches`

I don't have a fork of `opencv_extra`...

**Comment by alalek** [comment]
Ignore CN* builders. They have sporadic network issues.

opencv_extra/contrib forks are optional. This command just checks that.

**Comment by IanMaquignaz** [comment]
> 
> 
> Ignore CN* builders. They have sporadic network issues.
> 
> opencv_extra/contrib forks are optional. This command just checks that.

@alalek  Ok! Then unless you think there is merit to the [alternate unit test I wrote today](https://github.com/IanMaquignaz/opencv/blob/inverseRectification_newUnitTest_alternate/modules/calib3d/test/test_undistort.cpp), then I think it's good to go?

File: modules/calib3d/test/test_undistort.cpp [hunk 4]
```diff
@@ -1807,4 +1807,78 @@ TEST(Calib3d_initUndistortRectifyMap, regression_14467)
     EXPECT_LE(cvtest::norm(dst, mesh_uv, NORM_INF), 1e-3);
 }
 
+TEST(Calib3d_initInverseRectificationMap, regression_14468)
```
Comment (alalek) [inline modules/calib3d/test/test_undistort.cpp:1810]: > 14468

What does it mean?
Add PR/issue number. Use #20165 as reference on original code.

File: modules/calib3d/test/test_undistort.cpp [hunk 5]
```diff
@@ -1807,4 +1807,78 @@ TEST(Calib3d_initUndistortRectifyMap, regression_14467)
...
+            double cdist = 1./(1. + (d[0] + (d[1] + d[4]*r2)*r2)*r2); // (1. + (d[5] + (d[6] + d[7]*r2)*r2)*r2) == 1 as d[5-7]=0;
+            double x_ = (x - (d[2]*2.*x*y + d[3]*(r2 + 2.*x2)))*cdist;
+            double y_ = (y - (d[3]*2.*x*y + d[2]*(r2 + 2.*y2)))*cdist;
+
+            // Rectify
+            double X = R[0]*x_ + R[1]*y_ + R[2];
+            double Y = R[3]*x_ + R[4]*y_ + R[5];
+            double Z = R[6]*x_ + R[7]*y_ + R[8];
+            double x__ = X/Z;
+            double y__ = Y/Z;
```
Comment (alalek) [inline modules/calib3d/test/test_undistort.cpp:1873]: Perhaps it makes sense to add check that `Z` value doesn't drops to zero (`fabs(Z) > 1e-6`)

File: modules/calib3d/test/test_undistort.cpp (hunk 4 above)
Comment (IanMaquignaz) [reply modules/calib3d/test/test_undistort.cpp]: Haha, I incorrectly assumed that was an arbitrary number. Correction has been applied.

File: modules/calib3d/test/test_undistort.cpp (hunk 5 above)
Comment (IanMaquignaz) [reply modules/calib3d/test/test_undistort.cpp:1873]: This undistort function performs a rough estimate and should not be used as an undistort method. There are numerous weaknesses, including lack of regression (undistort should be iterative for best results) and missing handling for distortion coefficients outside of radial distortion.

As the coefficients are hard-coded to the unit test, I don't believe it is necessary to be overly thorough. If that is desired, full replication of cvUndistortPointsInternal() is necessary. This was attempted in the proposed [alternate unit test](https://github.com/IanMaquignaz/opencv/blob/inverseRectification_newUnitTest_alternate/modules/calib3d/test/test_undistort.cpp) but, aside from increasing the length of the code, did not add significant value.

**Review by alalek — APPROVED** [review]
Thank you :+1:

**Pull request merged**